providers/implementations/signature/rsa_sig.c.in

```diff
@@ -1503,9 +1503,7 @@ static const OSSL_PARAM *rsa_gettable_ctx_params(ossl_unused void *vprsactx,
 #ifdef FIPS_MODULE
 static int rsa_x931_padding_allowed(PROV_RSA_CTX *ctx)
 {
-    int approved = ((ctx->operation & EVP_PKEY_OP_SIGN) == 0);
-
-    if (!approved) {
+    if (ctx->operation & EVP_PKEY_OP_SIGNMSG || ctx->operation & EVP_PKEY_OP_SIGN) {
```

Comment:
This test was added before EVP_PKEY_OP_SIGNMSG existed, which is why it was missed.

Reply:
we caught it in a downstream test, I just ant to endure the fix is correct

Reply:
Updated